src/mongocxx/stdx.hpp

```diff
@@ -21,7 +21,7 @@ MONGOCXX_INLINE_NAMESPACE_BEGIN
 namespace stdx {
 
 // We adopt all the bsoncxx polyfills
-using namespace bsoncxx::stdx;
+using namespace ::bsoncxx::stdx;
```

Comment:
When I wrote this file, I felt a little bad. I wondered if it would have been better to have mongocxx/stdx/optional.hpp, etc., headers, but it just seemed like massive overkill.

However, I do wonder if we should be more restrictive here, and just import name by name as needed by mongocxx.

In other words, rather than saying using namespace in a header (generally not a great idea), just say

using ::bsoncxx::stdx::optional;
using ::bsoncxx::stdx::string_view;

etc.

What do people think?